lib/oauth2-client.js

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as storage from './storage';
+import * as bedrock from 'bedrock';
+import {httpClient} from '@digitalbazaar/http-client';
+import logger from './logger';
+const {util: {BedrockError}} = bedrock;
+
+export class OAuth2Client {
+/**
+ * Constructs an OAuth2Client instance to return an access token. It first
+ *  checks the database to see if an access token already exists for the client
+ *  id; if not it creates an access token then stores it in the database.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.config - The "oauth2-client-grant" properties.
+ * @param {object} options.httpsAgent - A NodeJS HTTPS Agent (`https.Agent`).
+ *
+ * @returns {OAuth2Client} The `OAuth2Client` instance.
+ */
+  constructor({config, httpsAgent} = {}) {
+    this.client_id = config.client_id;
+    this.client_secret = config.client_secret;
+    this.token_endpoint = config.token_endpoint;
+    this.grant_type = config.grant_type;
+    this.scope = config.scope;
+    this.maxRetries = config.maxRetries;
+    this.httpsAgent = httpsAgent;
+  }
+  /**
+   * Gets a new access token by first checking the database to see if an
+   *  access token already exists for the client id; if not it creates an
+   *  access token then stores it in the database.
+   *
+   * @returns {string} The access token.
+   */
+  async getToken() {
```

Comment:
There's an inconsistency in the API here where sometimes `Token` is used in the function name and other times `AccessToken` is. This should be normalized. I recommend using `AccessToken` since there are other types of tokens in oauth.

```suggestion
  async getAccessToken() {
```

Reply:
Agreed, updated.